### ISSUE TITLE
Godot 4 SDK

### ISSUE THREAD

**Issue by pietru2004** [icosa-foundation/open-brush#521]
I want to ask for godot 4 SDK that would allow to use sketches in godot... by that I mean ability to get most of brush materials working in godot... and reactivity to sound.

**Comment by mikeskydev** [comment]
Hi, while it would be awesome if a Godot SDK existed, we have no active contributors who are familiar with Godot at this time I'm afraid! If you know anyone, we'd happily try to help them build one.

**Comment by pietru2004** [comment]
I am familiar with godot gdscript, also gdextension allows other codding langs, and build with mono allows c# if I am correct, althrou I have almost no time. I could try it, but well, I am at school final class.

Well, mayby first, I should simply ask how advanced meshes are? Are there any custom shaders for animated materials? Etc. Does it need any GUI cause if not SDK could be just importer extension, plugin that says how to load custom res(code to load it)

**Comment by mikeskydev** [comment]
It really depends on the brush being used, a lot of strokes are basically triangle strips, but some are fully 3D tubes. Bascally all are custom shaders, though some can be PBR-approximated. I think the best way to tackle this would be to have a gdextension for Godot's glTF importer, that supports the extension Tilt Brush used to tag it's brushes: `GOOGLE_tilt_brush_material`. I built a toolkit for three.js that follows this pattern: https://github.com/icosa-foundation/three-icosa

**Comment by pietru2004** [comment]
Hm gona try looking at it later. It is hard to read code on the phone.

**Comment by pietru2004** [comment]
@mikeskydev so I think it would be easiest to just make plugin that only sets correct shader/material

**Comment by pietru2004** [comment]
Docs: https://docs.godotengine.org/en/latest/tutorials/plugins/editor/import_plugins.html
Demos: https://github.com/godotengine/godot-demo-projects/blob/master/plugins/addons/material_import_plugin

**Comment by mikeskydev** [comment]
Our primary export file format is glTF which has the specific identifiers you'd need to set the correct shader model, so I still think it makes sense to build the plugin as an addition to the glTF plugin. See https://github.com/godotengine/godot/pulls?q=is%3Apr+gltf+OMI for example adding some `OMI_` glTF extensions, or https://github.com/godotengine/godot/pull/78621

**Comment by pietru2004** [comment]
I know of GLTFDocumentExtension but I am not sure how to plug into default importer

**Comment by pietru2004** [comment]
@mikeskydev it is not like `*.tilt` files are gltf files are they ?

**Comment by mikeskydev** [comment]
No, but that's a much bigger project to build an importer for the file format as you would need to regenerate the meshes from raw control point data, which will be different depending on each brush. I think starting with GLTF will be good for sorting the shaders/materials part first, which you would need anyways for a theoretical .tilt importer

**Comment by pietru2004** [comment]
well I have one problem and that is absolutely 0 idea how to plug in into godot importer of gltf, like register function for checking materials and assign it, I have idea how to get materials and assign them, but not how to get to have it done on import...

I gona try experimenting with it... but well I am not sure won't I break something

**Comment by pietru2004** [comment]
hm... if materials are named on export wouldn't it be easiest to just provide godot ready shaders that could just be bound/applied to the objects? Color could be taken from vertex...
